src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -728,11 +728,14 @@ pImage = do
   let alt = fromAttrib "alt" tag
   let uid = fromAttrib "id" tag
   let cls = T.words $ fromAttrib "class" tag
+  let external = case fromAttrib "data-external" tag of
+                   "" -> []
+                   v  -> [("external", v)]
```

Comment:
Instead of handling this one attribute specially, why don't we use `toAttr` (defined in T.P.Readers.HTML.Parsing), as we do for many other tags?  This would carry over all the attributes, not just `external`.
Before calling `toAttr` we'd want to strip out `alt`, `src`, and `title`, since these are handled separately.  We could get rid of the special treatment of kvs on l. 734.

Reply:
Yes, that was much better. I've updated the PR with those changes.